### PR TITLE
migrate settings to json

### PR DESCRIPTION
**Pull request by mantikafasi**
replaces SettingsUtils to save plugin settings into /Aliucord/settings/pluginName.json instead of sharedpreferences ,also it transfers plugin settings from shared preferences to json on plugins first launch.

### CODE REVIEW

**Review by Vendicated — CHANGES_REQUESTED** [review]

File: Aliucord/src/main/java/com/aliucord/SettingsUtilsJSON.kt [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    }
+
+    private fun writeData() {
+        if (settings.length() > 0) {
+            val file = File(settingsFile)
+            try {
+                file.writeText(settings.toString(4))
+            } catch (e: Throwable) {
+                logger.info("Faled to save settings for $plugin")
+                logger.error(e)
```
[the remark this answers is not in the document]
Comment (Vendicated) [reply Aliucord/src/main/java/com/aliucord/SettingsUtilsJSON.kt:57]: this is not resolved

[the remark this answers is not in the document]
Comment (mantikafasi) [reply Aliucord/src/main/java/com/aliucord/SettingsUtilsJSON.kt]: no its resolved

File: Aliucord/src/main/java/com/aliucord/SettingsUtils.java [hunk 2]
```diff
@@ -154,4 +156,16 @@ public static void setObject(String key, Object val) {
         cache.put(key, val);
         setString(key, GsonUtils.toJson(val));
     }
+
+    public static void remove(String key){
```
Comment (Vendicated) [inline Aliucord/src/main/java/com/aliucord/SettingsUtils.java:160]: add space between ) and {, add javadoc

Comment (mantikafasi) [reply Aliucord/src/main/java/com/aliucord/SettingsUtils.java]: I hope its ok now

**Review by Vendicated — APPROVED** [review]

**Pull request merged**